play-services-base/core/src/main/AndroidManifest.xml

```diff
@@ -5,10 +5,17 @@
   -->
 <manifest xmlns:android="http://schemas.android.com/apk/res/android">
 
+    <permission android:name="org.microg.gms.permission.READ_SETTINGS"
+        android:protectionLevel="signature" />
+    <permission android:name="org.microg.gms.permission.WRITE_SETTINGS"
+        android:protectionLevel="signature" />
+
     <application>
         <provider
             android:name="org.microg.gms.settings.SettingsProvider"
             android:authorities="${applicationId}.microg.settings"
-            android:exported="false" />
+            android:exported="true"
+            android:readPermission="org.microg.gms.permission.READ_SETTINGS"
```

Comment:
Doesn't that imply that we need to add a `<uses-permission>` for this permission to `:play-services-base-core` itself, or we wouldn't be able to access the provider from within the app itself?

Reply:
Per [docs](https://developer.android.com/guide/topics/providers/content-provider-basics#Permissions), this is not necessary, as

> components in the provider's application always have full read and write access, regardless of the specified permissions.